x-pack/plugins/security/server/authentication/api_keys/api_keys.ts

```diff
@@ -335,6 +361,30 @@ export class APIKeys {
     return result;
   }
 
+  /**
+   * Tries to validate an API key.
+   * @param apiKeyPrams ValidateAPIKeyParams.
+   */
+  async validate(apiKeyPrams: ValidateAPIKeyParams): Promise<ValidateAPIKeyResult> {
```

Comment:
__note:__ I see @legrego suggested to add unit tests for the new `validate` function in https://github.com/elastic/kibana/pull/143867#pullrequestreview-1153491018. I agree with that too, would you mind adding a few unit tests (happy path, not happy path, disabled license)?